selfhost/parser.jakt

```diff
@@ -6,6 +6,7 @@
 //
 // SPDX-License-Identifier: BSD-2-Clause
 
+/// Expect: Skip
```

Comment:
🤔 this is uh. Kinda sus. We are putting expectations in the compiler source code? The tests and the compiler should not live in the same directory imo.

Reply:
Hmm, maybe instead of adding everything under selfhost we just add selfhost/main.jakt

Reply:
Adding selfhost dir to the tests is done so that Jakttest checks if selfhost can compile itself.